### PR TITLE
Remove application contexts when removing vocabulary fix

### PR DESCRIPTION
**Pull request by Mighantos**
Because I only tested #158 with Postman requests to SGOV running in IDE, I was not aware that when it built into a docker image, it won't work sometimes. Found out that when built as a docker image fetching the `applicationContexts` property in the `VocabularyContext` class, most of the time returned an empty list (despite working fine when running in IDE).

This PR fixes that by fetching application contexts by a native query. I tested it locally on my docker instance of the Assembly line.

Related #157

### CODE REVIEW

File: src/main/java/com/github/sgov/server/service/repository/VocabularyRepositoryService.java [hunk 1]
```diff
@@ -364,6 +364,6 @@ private void removeAllAttachments(VocabularyContext vocabularyContext) {
     }
 
     private void clearApplicationContexts(VocabularyContext vocabularyContext) {
```
Comment (blcham) [inline src/main/java/com/github/sgov/server/service/repository/VocabularyRepositoryService.java:366]: Little strange that calling public method from private one. Do we even need this method?

**Review by blcham — APPROVED** [review]
LGTM, although would be nicer to find out why previous approach did not work

File: src/main/java/com/github/sgov/server/service/repository/VocabularyRepositoryService.java (hunk 1 above)
Comment (Mighantos) [reply src/main/java/com/github/sgov/server/service/repository/VocabularyRepositoryService.java]: That's right we don't need it

**Comment by Mighantos** [comment]
About the fact why it did not work, I really don't know. I tried many things to make it work, from changing the order of clearing contexts linked to vocabulary context (to give it time to fetch the new application contexts) to changing the fetching strategy of the application contexts to lazy so it only loads them when needed. Everything without fixing the issue 100%. I also didn't see any discrepancies between my IDE runtime variables and docker image variables.

**Comment by blcham** [comment]
Ok, I don't see from the info where the issue can come from. This looks like issue related to cache-ing JOPA entities but at the same time I would expect to behave opposite way around (not working locally) ... The only bigger difference between docker and local development is JAVA version ... but also I would not expect that it brings this issue.

**Pull request merged**